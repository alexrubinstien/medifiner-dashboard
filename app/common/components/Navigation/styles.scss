@import './app/common/styles/utils/_vars.scss';

.navigation {
  padding: 30px 18px 24px;

  &__items-list {
    margin: 0;
    padding: 0;
    list-style: none;

    &:after {
      content: '';
      clear: both;
      display: table;
    }
  }

  &__item {
    position: relative;
    float: left;
    padding: 0 10px;
    font-size: 20px;
    font-weight: 500;
  }

  &__link {
    display: inline-block;
    padding: 2px 18px 4px;
    border-radius: 16px;
    color: $text-light-color;
    font-size: 16px;
    font-weight: 600;
    line-height: 26px;
    letter-spacing: 0.8px;
    text-decoration: none;
    transition: all .25s ease;

    &--active,
    &:hover {
      background: $primary-color-hover;
      color: #fff;
    }

    &--external:hover {
      background: none;
      color: $primary-color;
    }
  }

  &__external-link-icon {
    margin-left: 8px;
    color: $primary-color;
    font-size: 14px;
  }
}
