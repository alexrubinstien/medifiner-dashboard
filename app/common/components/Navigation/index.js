import React from 'react';
import { withRouter } from 'next/router';
import NextLink from 'next/link';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './styles.scss';

const Navigation = ({
  items,
  router,
}) => (
  <div className="navigation">
    <ul className="navigation__items-list">
      {items.map((item) => {
        let linkProps = {};
        if (item.external) {
          linkProps = {
            target: '_blank',
            rel: 'noopener noreferrer',
          };
        }

        return (
          <li
            key={item.href}
            className="navigation__item"
          >
            <NextLink href={item.href}>
              <a
                className={classNames('navigation__link', {
                  'navigation__link--active': router.pathname === item.href,
                  'navigation__link--external': item.external,
                })}
                {...linkProps}
              >
                {item.text}
                {item.external && <i className="fas fa-arrow-right navigation__external-link-icon" />}
              </a>
            </NextLink>
          </li>
        );
      })}
    </ul>
  </div>
);

Navigation.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    href: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  })).isRequired,
  router: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

export default withRouter(Navigation);
