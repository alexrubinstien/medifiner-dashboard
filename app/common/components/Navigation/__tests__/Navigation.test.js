/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Navigation from '../index.js';

describe('With Snapshot Testing', () => {
  it('Navigation component', () => {
    const component = renderer.create(
      <Navigation
        items={[{
          href: '/components',
          text: 'Map View',
        }, {
          href: '/historic-view',
          text: 'Historic View',
        }, {
          href: '/find-antiviral',
          text: 'Find an Antiviral',
        }]}
        router={{ pathname: '/components' }}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
