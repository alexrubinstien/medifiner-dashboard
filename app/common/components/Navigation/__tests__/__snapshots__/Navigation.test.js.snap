// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Navigation component 1`] = `
<div
  className="navigation"
>
  <ul
    className="navigation__items-list"
  >
    <li
      className="navigation__item"
    >
      <a
        className="navigation__link navigation__link--active"
        href="/components"
        onClick={[Function]}
      >
        Map View
      </a>
    </li>
    <li
      className="navigation__item"
    >
      <a
        className="navigation__link"
        href="/historic-view"
        onClick={[Function]}
      >
        Historic View
      </a>
    </li>
    <li
      className="navigation__item"
    >
      <a
        className="navigation__link"
        href="/find-antiviral"
        onClick={[Function]}
      >
        Find an Antiviral
      </a>
    </li>
  </ul>
</div>
`;
